internal/state/workspaces.go

```diff
@@ -20,11 +20,27 @@ type Workspaces struct {
 	repository      repository.RepositoryV2
 }
 
-func NewWorkspaces(repository repository.RepositoryV2) *Workspaces {
-	return &Workspaces{
+func NewWorkspaces(repository repository.RepositoryV2) (*Workspaces, error) {
+	// NOTE(Isaac799) if no workspaces exist we create one to prevent nil deref
+	// on startup when header getting selected workspace ddl option
+	workspaces, err := repository.LoadWorkspaces()
+	if err != nil {
+		return nil, err
+	}
+	if len(workspaces) == 0 {
+		workspace := domain.NewWorkspace("New Workspace")
```

Comment:
if you use domain.DefaultWorkspaceName as name then it will be by default protected from removal and edit. 

See: https://github.com/chapar-rest/chapar/blob/6a935d609107d9d7b8140e1ef846014ac40c3d5f/ui/pages/workspaces/view.go#L132

And https://github.com/chapar-rest/chapar/blob/6a935d609107d9d7b8140e1ef846014ac40c3d5f/ui/pages/workspaces/view.go#L158